// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.scp.core.common.proto;

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Global unique identifier.
message Uuid {
  uint64 high = 1; // The higher 64 bits.
  uint64 low = 2; // The lower 64 bits.
}

// Version number.
message Version {
  uint64 major = 1; // The major version number.
  uint64 minor = 2; // The minor version number.
}

// The status of execution result.
enum ExecutionStatus {
  EXECUTION_STATUS_UNSPECIFIED = 0; // Unknown status.
  EXECUTION_STATUS_SUCCESS = 1; // The execution is successful.
  EXECUTION_STATUS_FAILURE = 2; // The execution is failed.
  EXECUTION_STATUS_RETRY = 3; // The execution is retriable.
}

// The result of the execution.
message ExecutionResult {
  ExecutionStatus status = 1; // The execution status.
  // The failure status code.
  uint64 status_code = 2;
  // Returned if the execution failed.
  string error_message = 3;
}
